src/main/java/com/hivemq/cli/commands/options/TlsOptions.java

```diff
@@ -138,85 +137,298 @@ private boolean useBuiltSslConfig() {
                         Throwables.getRootCause(e).getMessage());
+                throw e;
             }
-        }
-
-        if (clientPrivateKey == null) {
+            final PrivateKey privateKey;
+            final String defaultClientPrivateKeyPassword = defaultCLIProperties.getClientPrivateKeyPassword();
             try {
-                clientPrivateKey = defaultCLIProperties.getClientPrivateKey();
+                privateKey = tlsUtil.getPrivateKeyFromFile(defaultClientPrivateKey, defaultClientPrivateKeyPassword);
             } catch (final Exception e) {
                 Logger.error(e,
                         "Default client private key could not be loaded ({})",
                         Throwables.getRootCause(e).getMessage());
+                throw e;
             }
+            final String password = "filler";
+            final KeyStore ks = KeyStore.getInstance(KeyStore.getDefaultType());
+            ks.load(null, null);
+            ks.setKeyEntry("mykey",
+                    privateKey,
+                    password.toCharArray(),
+                    x509Certificates.toArray(new X509Certificate[0]));
+
+            keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            keyManagerFactory.init(ks, password.toCharArray());
+            //nothing found
+        } else {
+            keyManagerFactory = null;
         }
+        return keyManagerFactory;
+    }
 
-        try {
-            final Collection<X509Certificate> defaultServerCertificate =
-                    defaultCLIProperties.getServerCertificateChain();
-            if (defaultServerCertificate != null) {
-                if (serverCertificateChain == null) {
-                    serverCertificateChain = new ArrayList<>();
+    private @NotNull KeyManagerFactory createKeyManagerFactoryFromKeystore(
+            final @NotNull Path clientKeystore,
+            final @Nullable String clientKeystorePassword,
+            final @Nullable String clientKeystorePrivateKeyPassword)
+            throws NoSuchAlgorithmException, KeyStoreException, IOException, CertificateException,
+            UnrecoverableKeyException {
+        final KeyManagerFactory keyManagerFactory =
+                KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm());
+        if (TlsUtil.isPKCS12Keystore(clientKeystore)) {
+            final KeyStore keyStore = KeyStore.getInstance("PKCS12");
+            final char[] keystorePassword;
+            if (clientKeystorePassword != null) {
+                keystorePassword = clientKeystorePassword.toCharArray();
+            } else {
+                keystorePassword = PasswordUtils.readPassword("Enter keystore password:");
+            }
+            try (final InputStream inputStream = Files.newInputStream(clientKeystore, StandardOpenOption.READ)) {
+                keyStore.load(inputStream, keystorePassword);
```

Comment:
maybe move this into a method to return password and load keystore to avoid the duplications